.teamcity/Gradle_Check/configurations/PerformanceTest.kt

```diff
@@ -25,25 +20,43 @@ class PerformanceTest(model: CIBuildModel, type: PerformanceTestType, stage: Sta
         }
     }
 
+    requirements {
+        // We observed some relocatability issues with its inputs and it is a long running job
+        doesNotContain("teamcity.agent.name", "ec2")
+    }
+
     params {
         param("performance.baselines", type.defaultBaselines)
     }
 
-    steps {
+    fun BuildSteps.runner(runnerName: String, runnerTasks: String, extraParameters: String = "") {
         gradleWrapper {
-            name = "GRADLE_RUNNER"
-            tasks = ""
-            gradleParams = (
-                    buildToolGradleParameters(isContinue = false)
-                        + performanceTestCommandLine(task = "distributed${type.taskId}s", baselines = "%performance.baselines%", extraParameters = type.extraParameters)
-                        + distributedPerformanceTestParameters(IndividualPerformanceScenarioWorkers(model).id.toString())
-                        + listOf(buildScanTag("PerformanceTest"))
-                        + model.parentBuildCache.gradleParameters(Os.linux)
-                    ).joinToString(separator = " ")
+            name = runnerName
+            tasks = "clean ${runnerTasks}"
+            gradleParams = (buildToolGradleParameters(isContinue = false)
+                + performanceTestParameters(baselines = "%performance.baselines%", extraParameters = type.extraParameters)
+                + distributedPerformanceTestParameters(IndividualPerformanceScenarioWorkers(model).id.toString())
+                + listOf(buildScanTag("PerformanceTest"))
+                + model.parentBuildCache.gradleParameters(Os.linux)
+                + extraParameters
+                ).joinToString(separator = " ")
         }
+    }
+
+    steps {
+        runner("GRADLE_RUNNER", "distributed${type.taskId}s")
         checkCleanM2()
-        tagBuild(model, true)
+        if (type.hasRerunner) {
+            val rerunnerParameters = listOf(
+                "-PteamCityBuildId=%teamcity.build.id%",
+                "-PonlyPreviousFailedTestClasses=true",
+                "-PgithubToken=%github.ci.oauth.token%")
+            runner("GRADLE_RERUNNER", "tagBuild distributed${type.taskId}s", rerunnerParameters.joinToString(" "))
```

Comment:
we need to use `tagBuild` first, since the arguments to the `distributedPerformanceTests` is coming in the `gradleParams`?